### PR TITLE
cs_template: implement update and revamp

### PR DESCRIPTION
**Pull request by resmo**
##### SUMMARY
- Implement update feature
- Add the "all" filter
- Deprecated is_ready param (no function)
- Add tests (does not run in simulator [yet?])
- change the way we find templates

I changed a bit how we find templates and would like to get some feedback.

Previously, the module assumed, that the template name is unique. However, some cloud providers (e.g. exoscale) use name just like an "os type" and use displaytext with the specific name.

~~~json
[
    {
      "account": "exostack", 
      "checksum": "2ce52685a7a4eb3031cd4c0b0c452336", 
      "created": "2017-10-09T15:24:40+0200", 
      "crossZones": false, 
      "displaytext": "Linux Debian 9 64-bit 10G Disk (2017-10-08-622866)", 
      "domain": "ROOT", 
      "domainid": "4a8857b8-7235-4e31-a7ef-b8b44d180850", 
      "format": "QCOW2", 
      "hypervisor": "KVM", 
      "id": "02144e63-280c-466f-9dd2-37b13bb1e014", 
      "isdynamicallyscalable": false, 
      "isextractable": false, 
      "isfeatured": false, 
      "ispublic": true, 
      "isready": true, 
      "name": "Linux Debian 9 64-bit", 
      "ostypeid": "113038d0-a8cd-4d20-92be-ea313f87c3ac", 
      "ostypename": "Other PV (64-bit)", 
      "passwordenabled": true, 
      "size": 10737418240, 
      "sshkeyenabled": false, 
      "tags": [], 
      "templatetype": "USER", 
      "zoneid": "4da1b188-dcd6-4ff5-b7fd-bde984055548", 
      "zonename": "at-vie-1"
    }, 
    {
      "account": "exostack", 
      "checksum": "aa8f45d100f5e99492869907be9cf996", 
      "created": "2017-10-09T15:24:40+0200", 
      "crossZones": false, 
      "displaytext": "Linux Debian 9 64-bit 200G Disk (2017-10-08-622866)", 
      "domain": "ROOT", 
      "domainid": "4a8857b8-7235-4e31-a7ef-b8b44d180850", 
      "format": "QCOW2", 
      "hypervisor": "KVM", 
      "id": "9ae8ce90-d3b5-4c47-8f05-6123fbe1234f", 
      "isdynamicallyscalable": false, 
      "isextractable": false, 
      "isfeatured": false, 
      "ispublic": true, 
      "isready": true, 
      "name": "Linux Debian 9 64-bit", 
      "ostypeid": "113038d0-a8cd-4d20-92be-ea313f87c3ac", 
      "ostypename": "Other PV (64-bit)", 
      "passwordenabled": true, 
      "size": 214748364800, 
      "sshkeyenabled": false, 
      "tags": [], 
      "templatetype": "USER", 
      "zoneid": "4da1b188-dcd6-4ff5-b7fd-bde984055548", 
      "zonename": "at-vie-1"
    }
 ]
}
~~~

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
cs_template

##### ANSIBLE VERSION

```
2.6
```

##### ADDITIONAL INFORMATION

@marcaurele @dpassante @vincentbernat  I would be interested in your thoughts about this change, especially the way I changed how to find an existing template. Before I looked for the name to match, so the name had to be unique. 

With this change, if the name is not unique, the user registering templates can add display_text as a "search option" so this idempotency is given.

### CODE REVIEW

**Comment by marcaurele** [comment]
@resmo I like the idea to be able to match on either the name or the displaytext. It will be helpful to have this choice. I don't see in your tests a usecase with `template_find_option` but I guess it's still *wip*.

**Comment by resmo** [comment]
build depends on https://github.com/ansible/cloudstack-test-container/pull/5

**Comment by resmo** [comment]
I am happy with it. Any reviewers?

ready_for_review

**Comment by dpassante** [comment]
@resmo Very nice enhancement and the ```template_find_options``` param approach seems to make possible to update either the ```name``` or the ```display text``` of a template, which is an useful feature too.
The code LGTM, I'll try to make some tests tomorrow.

File: test/integration/targets/cs_template/tasks/test1.yml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  register: template
+- name: verify test download template idempotence
+  assert:
+    that:
+    - template is not changed
+    - template.name == "ansible-template-test1"
+    - template.display_text == "ansible-template-test1"
+    - template.cross_zones == true
+
+- name: test update template in check mdoe
```
Comment (marcaurele) [inline test/integration/targets/cs_template/tasks/test1.yml]: nitpick: `in check mode`

**Review by marcaurele — APPROVED** [review]
LGTM

**Comment by resmo** [comment]
a quick reminder: 2018-08-30 Community Freeze (Non-Core Modules/Plugins). Would be great if I get feedback asap but before then

**Review by dpassante — APPROVED** [review]
All my manual tests are passed.
Nice job ;)

**Comment by resmo** [comment]
please comment with shipit then :)

**Comment by dpassante** [comment]
sorry :)

**Comment by dpassante** [comment]
shipit

**Comment by dpassante** [comment]
shipit

**Pull request merged**